src/store/modules/lib/agents.js

```diff
@@ -57,11 +60,91 @@ export default () => ({
 		[UPDATE_USER_AGENT_TEXT] (state, { userAgentId, text }) {
 			state.userAgents.find(a => a.id === userAgentId).text = text
 		},
-		[CREATE_USER_AGENT] (state, { id, text, label }) {
-			state.userAgents.unshift({
-				id,
-				label,
-				text
+		[CREATE_USER_AGENT] (state, { id, text, label, isShared }) {
+			if (!isShared) {
+				state.userAgents.unshift({
+					id,
+					label,
+					text
+				})
+			} else {
+				let findOldAgentText = state.userAgents.find(a => a.text === text)
+				let findOldAgentLabel = state.userAgents.find(a => a.label === label)
+				if (findOldAgentText) {
+					findOldAgentText.text = text
+					findOldAgentText.id = id
+					// This below condition checks. If there is already named label and the old text and if that text === label's state text
+				} else if (findOldAgentLabel && findOldAgentText && findOldAgentLabel.text === findOldAgentText.text) {
```

Comment:
this in never true after `if (findOldAgentText)` above

Reply:
yes, good catch should have followed this.